metal3-dev/start_local_ironic.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+source $(dirname $(dirname $0))/logging.sh
+source $(dirname $(dirname $0))/common.sh
+source $(dirname $(dirname $0))/utils.sh
+
+DHCP_RANGE=${DHCP_RANGE:-172.22.0.10,172.22.0.100}
+
+# Add firewall rules to ensure the IPA ramdisk can reach Ironic and the Inspector API on the host
+
+for port in 5050 6385 ; do
+    if [ "${RHEL8}" = "True" ] ; then
+        sudo firewall-cmd --zone=libvirt --add-port=${port}/tcp
+        sudo firewall-cmd --zone=libvirt --add-port=${port}/tcp --permanent
+    else
+        if ! sudo iptables -C INPUT -i provisioning -p tcp -m tcp --dport $port -j ACCEPT > /dev/null 2>&1; then
+            sudo iptables -I INPUT -i provisioning -p tcp -m tcp --dport $port -j ACCEPT
+        fi
+    fi
+done
+
+# Allow access to dhcp and tftp server for pxeboot
+
+for port in 67 69 ; do
+    if [ "${RHEL8}" = "True" ] ; then
+        sudo firewall-cmd --zone=libvirt --add-port=${port}/udp
+        sudo firewall-cmd --zone=libvirt --add-port=${port}/udp --permanent
+    else
+        if ! sudo iptables -C INPUT -i provisioning -p udp --dport $port -j ACCEPT 2>/dev/null ; then
+            sudo iptables -I INPUT -i provisioning -p udp --dport $port -j ACCEPT
+        fi
+    fi
+done
+
+mariadb_password=$(echo $(date;hostname)|sha256sum |cut -c-20)
+
+# Start dnsmasq, http, mariadb, and ironic containers using same image
+
+sudo podman run -d --net host --privileged --name dnsmasq  --pod ironic-pod \
+     -v $IRONIC_DATA_DIR:/shared --entrypoint /bin/rundnsmasq \
+     --env DHCP_RANGE="$DHCP_RANGE" ${IRONIC_IMAGE}
+
+sudo podman run -d --net host --privileged --name mariadb --pod ironic-pod \
+     -v $IRONIC_DATA_DIR:/shared --entrypoint /bin/runmariadb \
+     --env MARIADB_PASSWORD=$mariadb_password ${IRONIC_IMAGE}
+
+sudo podman run -d --net host --privileged --name ironic-conductor --pod ironic-pod \
+     --env MARIADB_PASSWORD=$mariadb_password \
+     --env OS_CONDUCTOR__HEARTBEAT_TIMEOUT=120 \
+     --entrypoint /bin/runironic-conductor \
+     -v $IRONIC_DATA_DIR:/shared ${IRONIC_IMAGE}
+
+sudo podman run -d --net host --privileged --name ironic-api --pod ironic-pod \
+     --env MARIADB_PASSWORD=$mariadb_password \
+     --entrypoint /bin/runironic-api \
+     -v $IRONIC_DATA_DIR:/shared ${IRONIC_IMAGE}
+
+# Start Ironic Inspector
+sudo podman run -d --net host --privileged --name ironic-inspector \
+     --pod ironic-pod -v $IRONIC_DATA_DIR:/shared "${IRONIC_INSPECTOR_IMAGE}"
+
+# Make sure Ironic is up
+export OS_URL=http://localhost:6385
+
+wait_for_json ironic \
+    "${OS_URL}/v1/nodes" \
+    20 \
+    -H "Accept: application/json"
+
+if [ $(sudo podman ps | grep -w -e "ironic-api$" -e "ironic-conductor$" -e "ironic-inspector$" -e "dnsmasq" -e "httpd" | wc -l) != 5 ]; then
+    echo "Can't find required containers"
+    exit 1
+fi
+
+mkdir -p ~/.config/openstack
+cat > ~/.config/openstack/clouds.yaml <<EOF
```

Comment:
Doesn't this wipe out a user's existing config? Could we just put it in dev-scripts root since the openstack client looks in $PWD?

Reply:
I'm not entirely sure we should keep existing configs, this setup is not compatible with them.. Using PWD will lead to confusion when a user moves to another directory. I can probably write a Python script that merges clouds.yaml if it's deemed useful.

Reply:
Please don't overwrite my user files. I have a couple of things in there, not just dev-scripts. I agree with Stephen that it would be better to modify `clouds.yaml` that we check in to dev-scripts.

Reply:
Right, I was just thinking people might have other Ironics they're configured to talk to, not just dev-scripts ones.

The dev-scripts one actually gets generated every time dynamically (since provisioningNetworkCIDR is configurable) so you can just change the path to be dev-scripts root.

Reply:
Since we already maintain a clouds.yaml which does depend on $PWD folks should be used to that - so perhaps we can just add a "local" option to that selected via OS_CLOUD?